packages/services/api/src/modules/audit-logs/module.graphql.ts

```diff
@@ -0,0 +1,361 @@
+  type TargetDeletedAuditLog implements AuditLog {
+    id: ID!
+    eventTime: Date!
+    userId: String!
+    userEmail: String!
+    organizationId: String!
+    eventType: String!
+    metadata: TargetDeletedAuditLogMetadata!
+  }
+
+  type TargetDeletedAuditLogMetadata {
+    projectId: String!
+    targetId: String!
+    targetName: String!
+    readableMessage: String!
+  }
+
+  type SchemaPolicySettingsUpdatedAuditLog implements AuditLog {
+    id: ID!
+    eventTime: Date!
+    userId: String!
+    userEmail: String!
+    organizationId: String!
+    eventType: String!
+    metadata: SchemaPolicySettingsUpdatedAuditLogMetadata!
+  }
+
+  type SchemaPolicySettingsUpdatedAuditLogMetadata {
+    projectId: String!
+    updatedFields: JSON!
+    readableMessage: String!
+  }
+
+  type SchemaCheckedAuditLog implements AuditLog {
+    id: ID!
+    eventTime: Date!
+    userId: String!
+    userEmail: String!
+    organizationId: String!
+    eventType: String!
+    metadata: SchemaCheckedAuditLogMetadata!
+  }
+
+  type SchemaCheckedAuditLogMetadata {
+    projectId: String!
+    schemaId: String!
+    schemaName: String!
+    readableMessage: String!
+  }
+
+  type SchemaPublishAuditLog implements AuditLog {
+    id: ID!
+    eventTime: Date!
+    userId: String!
+    userEmail: String!
+    organizationId: String!
+    eventType: String!
+    metadata: SchemaPublishAuditLogMetadata!
+  }
+
+  type SchemaPublishAuditLogMetadata {
+    projectId: String!
+    schemaId: String!
+    schemaName: String!
+    readableMessage: String!
+  }
+
+  type SchemaDeletedAuditLog implements AuditLog {
+    id: ID!
+    eventTime: Date!
+    userId: String!
+    userEmail: String!
+    organizationId: String!
+    eventType: String!
+    metadata: SchemaDeletedAuditLogMetadata!
+  }
+
+  type SchemaDeletedAuditLogMetadata {
+    projectId: String!
+    schemaId: String!
+    schemaName: String!
+    readableMessage: String!
+  }
+
+  type RoleCreatedAuditLog implements AuditLog {
+    id: ID!
+    eventTime: Date!
+    userId: String!
+    userEmail: String!
+    organizationId: String!
+    eventType: String!
+    metadata: RoleCreatedAuditLogMetadata!
+  }
+
+  type RoleCreatedAuditLogMetadata {
+    projectId: String!
+    roleId: String!
+    roleName: String!
+    readableMessage: String!
+  }
+
+  type RoleAssignedAuditLog implements AuditLog {
+    id: ID!
+    eventTime: Date!
+    userId: String!
+    userEmail: String!
+    organizationId: String!
+    eventType: String!
+    metadata: RoleAssignedAuditLogMetadata!
+  }
+
+  type RoleAssignedAuditLogMetadata {
+    projectId: String!
+    roleId: String!
+    roleName: String!
+    userIdAssigned: String!
+    userEmailAssigned: String!
+    readableMessage: String!
+  }
+
+  type RoleDeletedAuditLog implements AuditLog {
+    id: ID!
+    eventTime: Date!
+    userId: String!
+    userEmail: String!
+    organizationId: String!
+    eventType: String!
+    metadata: RoleDeletedAuditLogMetadata!
```

Comment:
maybe you don't need `metadata: ` field in each type? why not just add the fields in `RoleDeletedAuditLog`? it will cut the amount of new types by /2 😆